homeassistant/components/ads/sensor.py

```diff
@@ -58,21 +77,41 @@ def setup_platform(
     name = config[CONF_NAME]
     unit_of_measurement = config.get(CONF_UNIT_OF_MEASUREMENT)
     factor = config.get(CONF_ADS_FACTOR)
+    device_class_str = config.get("device_class")
 
-    entity = AdsSensor(ads_hub, ads_var, ads_type, name, unit_of_measurement, factor)
+    device_class = SensorDeviceClass(device_class_str) if device_class_str else None
+
+    entity = AdsSensor(
+        ads_hub, ads_var, ads_type, name, unit_of_measurement, factor, device_class
+    )
 
     add_entities([entity])
 
 
 class AdsSensor(AdsEntity, SensorEntity):
     """Representation of an ADS sensor entity."""
 
-    def __init__(self, ads_hub, ads_var, ads_type, name, unit_of_measurement, factor):
+    def __init__(
+        self,
+        ads_hub,
+        ads_var,
+        ads_type,
+        name,
+        unit_of_measurement,
+        factor,
+        device_class,
+    ):
```

Comment:
Can we add more types in a follow up PR?

Reply:
what do you mean? if ADS_TYPES, all are here except string. That doesn't makes sense (to me). :)

Reply:
No I mean, python typing, so adding `name: str` to make the code and linters know that `name` is of type `str`

Reply:
added them as you requested.